core/src/main/java/com/scalar/db/storage/dynamo/DynamoAdmin.java

```diff
@@ -86,20 +343,334 @@ public TableMetadata getTableMetadata(String namespace, String table) throws Exe
+                getGlobalIndexResourceID(namespace, table, secondaryIndex), scalingType));
+      }
+    }
+
+    // request
+    for (RegisterScalableTargetRequest registerScalableTargetRequest :
+        registerScalableTargetRequestList) {
+      try {
+        applicationAutoScalingClient.registerScalableTarget(registerScalableTargetRequest);
+      } catch (ApplicationAutoScalingException e) {
+        throw new ExecutionException(
+            "Unable to register scalable target for " + registerScalableTargetRequest.resourceId(),
+            e);
+      }
+    }
+
+    for (PutScalingPolicyRequest putScalingPolicyRequest : putScalingPolicyRequestList) {
+      try {
+        applicationAutoScalingClient.putScalingPolicy(putScalingPolicyRequest);
+      } catch (ApplicationAutoScalingException e) {
+        throw new ExecutionException(
+            "Unable to put scaling policy request for " + putScalingPolicyRequest.resourceId(), e);
+      }
+    }
+  }
+
+  public void disableAutoScaling(String namespace, String table) {
+    TableMetadata tableMetadata = metadataManager.getTableMetadata(namespace, table);
+    if (tableMetadata == null) {
+      return;
+    }
+    Set<String> secondaryIndexes = tableMetadata.getSecondaryIndexNames();
+    List<DeregisterScalableTargetRequest> deregisterScalableTargetRequestList = new ArrayList<>();
+    List<DeleteScalingPolicyRequest> deleteScalingPolicyRequestList = new ArrayList<>();
+
+    // write, read scaling of table
+    for (String scalingType : TABLE_SCALING_TYPE_SET) {
+      deregisterScalableTargetRequestList.add(
+          buildDeregisterScalableTargetRequest(getTableResourceID(namespace, table), scalingType));
+      deleteScalingPolicyRequestList.add(
+          buildDeleteScalingPolicyRequest(getTableResourceID(namespace, table), scalingType));
+    }
+
+    // write, read scaling of global indexes (secondary indexes)
+    for (String secondaryIndex : secondaryIndexes) {
+      for (String scalingType : SECONDARY_INDEX_SCALING_TYPE_SET) {
+        deregisterScalableTargetRequestList.add(
+            buildDeregisterScalableTargetRequest(
+                getGlobalIndexResourceID(namespace, table, secondaryIndex), scalingType));
+        deleteScalingPolicyRequestList.add(
+            buildDeleteScalingPolicyRequest(
+                getGlobalIndexResourceID(namespace, table, secondaryIndex), scalingType));
+      }
+    }
+
+    // request
+    for (DeleteScalingPolicyRequest deleteScalingPolicyRequest : deleteScalingPolicyRequestList) {
+      try {
+        applicationAutoScalingClient.deleteScalingPolicy(deleteScalingPolicyRequest);
+      } catch (ApplicationAutoScalingException e) {
+        if (!(e instanceof ObjectNotFoundException)) {
+          LOGGER.warn(
+              "Delete scaling policy " + deleteScalingPolicyRequest.policyName() + " failed. " + e);
+        }
+      }
+    }
+    for (DeregisterScalableTargetRequest deregisterScalableTargetRequest :
```

Comment:
Please put an empty line above.

Reply:
Apply in [391f30b](https://github.com/scalar-labs/scalardb/commit/391f30b54026c24a765bd56b672570feca195ed3)